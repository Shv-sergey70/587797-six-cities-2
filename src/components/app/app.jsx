import React from 'react';
import MainPage from "../main-page/main-page";
import OfferDetail from "../offer-detail/offer-detail";
import {Switch, Route} from 'react-router-dom';
import SignIn from "../sign-in/sign-in";
import {withAuthForm} from "../../hocs/with-auth-form/with-auth-form";
import {Route as AppRoute} from '../../const/routes';
import FavoritesPage from "../favorites-page/favorites-page";
import Operation from "../../operation";
import {connect} from "react-redux";
import PropTypes from "prop-types";

const SignInWrapped = withAuthForm(SignIn);

class App extends React.PureComponent {
  componentDidMount() {
    const {
      checkAuth
    } = this.props;

    checkAuth();
  }

  render() {
    return <Switch>
      <Route path={AppRoute.MAIN} exact component={MainPage}/>
      <Route path={AppRoute.LOGIN} exact component={SignInWrapped}/>
      <Route path={AppRoute.FAVORITES} exact component={FavoritesPage}/>
      <Route path={AppRoute.OFFER} exact component={OfferDetail}/>
      <Route
        render={() => (
          <h1>404</h1>
        )}/>
    </Switch>;

    // if (/^\/offer\/\d$/.test(location.pathname)) {
    //   const offerId = Number(location.pathname.match(/^\/offer\/(\d)$/)[1]);
    //   const selectedOffer = allOffers.find((offer) => offer.id === offerId);
    //
    //   if (selectedOffer !== undefined) {
    //     return <OfferDetail offer={selectedOffer} />;
    //   }
    // }
  }
}

App.propTypes = {
  checkAuth: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  checkAuth: () => dispatch(Operation.checkAuth()),
});

export {App};

export default connect(null, mapDispatchToProps)(App);
