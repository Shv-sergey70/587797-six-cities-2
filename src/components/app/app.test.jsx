import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from "react-router-dom";
import {App} from "./app";

jest.mock(`../main-page/main-page.jsx`, () => `MainPage`);
jest.mock(`../sign-in/sign-in.jsx`, () => `SignIn`);
jest.mock(`../offer-detail/offer-detail.jsx`, () => `OfferDetail`);
jest.mock(`../favorites-page/favorites-page.jsx`, () => `FavoritesPage`);

it(`Main page with offer renders correctly`, () => {
  const tree = renderer.create(<Router>
    <App
      checkAuth={jest.fn()}
    />
  </Router>).toJSON();

  expect(tree).toMatchSnapshot();
});
