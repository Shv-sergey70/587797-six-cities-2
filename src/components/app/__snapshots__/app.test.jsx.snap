// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders 1`] = `
<Connect(MainPage)
  allOffers={
    Array [
      Object {
        "city": Object {
          "name": "Paris",
        },
        "id": 1,
        "isFavorite": true,
        "isPremium": true,
        "previewImage": "/img/apartment-01.jpg",
        "price": 80,
        "rating": 4.5,
        "title": "Beautiful & luxurious apartment at great location",
        "type": "Apartment",
      },
      Object {
        "city": Object {
          "name": "Brussels",
        },
        "id": 2,
        "isFavorite": false,
        "isPremium": true,
        "previewImage": "/img/apartment-01.jpg",
        "price": 180,
        "rating": 4,
        "title": "Wood and stone place",
        "type": "Private room",
      },
      Object {
        "city": Object {
          "name": "Paris",
        },
        "id": 3,
        "isFavorite": true,
        "isPremium": false,
        "previewImage": "/img/apartment-02.jpg",
        "price": 120,
        "rating": 5,
        "title": "Canal View Prinsengracht",
        "type": "Apartment",
      },
      Object {
        "city": Object {
          "name": "Amsterdam",
        },
        "id": 4,
        "isFavorite": false,
        "isPremium": false,
        "previewImage": "/img/apartment-03.jpg",
        "price": 40,
        "rating": 3,
        "title": "Nice, cozy, warm big bed apartment",
        "type": "Private room",
      },
    ]
  }
/>
`;
